### DISCUSSION TITLE
Crosspost from /WSL2-Linux-Kernel - .wslconfig "processors"

### DISCUSSION THREAD

**Discussion by just-Bri** [microsoft/WSL#5943] [General]
https://github.com/microsoft/WSL2-Linux-Kernel/issues/188

So there is some basic documentation for the wslconfig options available. One thing I am not sure about is what "processors" actually relates to. I have a 4 core 8 thread cpu, when I set the wslconfig processor option to 2, does that assign 2 physical cpu cores to the VM? Or does that allow 2 threads to the VM?

**Comment by therealkenc** [comment]
Documentation for `.wslconfig` is [here](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig).

~Which SMT thread goes to work is up to the Windows scheduler. Shorter, simpler answer: you get two virtual processors, which you can see in `/proc/cpuinfo`. Neither threads nor cores are "assigned".~

**Reply by just-Bri** [reply]
I'm trying to figure out what is being used on the hardware side. If you look at my other issue from WSL2-Linux-Kernel you'll see what I mean. And I even have a screenshot from the documentation in that thread where it says "processors" but does not define whether or not that is a core or a thread. 

/proc/cpuinfo only tells me what the VM knows about, but it doesn't know how many physical cores, or threads, on the hardware CPU it is using.

**Reply by therealkenc** [reply]
Was curious so did a test to look. The logical CPUs aren't being scheduled evenly, either because they are (as a practical matter) assigned at startup or because there is some strong processor affinity going on. On my 16C/32T box with `processors=8`, I ended roughly the same logical processors doing the heaving lifting on two runs of "`sysbench cpu --threads=8 run`". Either way the work was not allocated by physical cores vs logical cores, AFAICT. So short answer, as a practical matter, "processors" means "logical processors". Which logical processors is implementation detail.

Run 1:

![image](https://user-images.githubusercontent.com/7049700/93541220-a5827180-f90a-11ea-8f53-d2bf7691946c.png)

Run 2:

![image](https://user-images.githubusercontent.com/7049700/93541166-7f5cd180-f90a-11ea-8d00-a09a10ea96bb.png)

**Reply by maltem-za** [reply]
Just wanted to add that @therealkenc's answer is confirmed by a comment in the [example .wslconfig file](https://docs.microsoft.com/en-us/windows/wsl/wsl-config#example-wslconfig-file): "Sets the VM to use two **virtual processors**"
